tracer/tracer.go

```diff
@@ -52,6 +55,12 @@ func NewTracer() *Tracer {
 
 // NewTracerTransport create a new Tracer with the given transport.
 func NewTracerTransport(transport Transport) *Tracer {
+	randSource, err := newRandSource()
+	if err != nil {
+		panic(fmt.Sprintf("cannot create random source: %v", err))
```

Comment:
I'm a little worried with panicking here, if, for some reason, creating the rand source we should probably log it, make it blatant there's a problem, and make all tracing no-op so that no span/trace are sent, making it clear something is going wrong. If we do panic here for real, we might stop user code and create an outage on our behalf. Generally speaking, panicking on totally unexpected thing is OK, but here we're hooking ourselves on existing code, trying to be as furtive as possible.